script/check-simulation-build-cmake

```diff
@@ -42,6 +42,13 @@ build_all_features()
     reset_source
     "$(dirname "$0")"/cmake-build simulation -DOT_THREAD_VERSION=1.1
 
+    # Build Thread 1.1 cli-radio
+    # cli-radio should be built alone because in a full build, `libopenthread-radio` should be built with
+    # `OPENTHREAD_CONFIG_DIAG_NATIVE_CMDS_ON_RCP=0`. While cli-radio needs `libopenthread-radio` built with
+    # `OPENTHREAD_CONFIG_DIAG_NATIVE_CMDS_ON_RCP=1`
```

Comment:
Is there a way to remove this config? For example, can we always build the native diag code but rely on linking and/or initialization to actually enable it?

Reply:
Sounds good to me. I will try to do that.

Reply:
After some trying and re-thinking, I think:
* It may not be good to always build both diag code and choose which one to use at runtime
    * For a usual RCP, the native diag code is totally useless. On some SoCs, our RCP application FW just fits in the size limit. So we should avoid adding code to normal RCP as possible.
    * This makes the code logic more complex. For example, we may need to add a flag in the Diags module to tell if it should use the native diag code or RCP diag code. And this applys to both Diags modules (Host and RCP). That says, we need to ensure the flag is correct on both host and RCP. We can use configuration as a default value for the flag. But this goes back, we still add one more config. 
* I tried a linking solution: move different diag code to two new source files and build two different `openthread-radio` library (most things are common except for some diag and radio_callback code). This should work. One flaw is that those class members (only for native diag code) will be there anyway. (I cannot think of a way to avoid this, is there any trick to do that without adding any configuration?)
* I think the best way is to add a new build type for openthread core (For example, `OPENTHREAD_RADIO_ALONE`, which is a sibling of `OPENTHREAD_FTD`, `OPENTHREAD_MTD` and `OPENTHREAD_RADIO`). And we will also build a different lib `openthread-radio-alone` and let ot-cli-rcp link this lib. This is much simpler.

Thoughts? @jwhui

Reply:
Having another build type/lib makes sense. However, I'm open to others opinions on the preferred approach. Thoughts from others?